components/director/internal/domain/application/service.go

```diff
@@ -256,14 +257,11 @@ func (s *service) Create(ctx context.Context, in model.ApplicationRegisterInput)
 	if err != nil {
 		return "", err
 	}
+	s.scenariosService.AddDefaultScenarioIfEnabled(&in.Labels)
 
-	if _, ok := in.Labels[model.ScenariosKey]; !ok {
-		if in.Labels == nil {
-			in.Labels = map[string]interface{}{}
-		}
-		in.Labels[model.ScenariosKey] = model.ScenariosDefaultValue
+	if in.Labels == nil {
```

Comment:
Hmm, if we need to initialize labels anyway, can we do it before `AddDefaultScenarioIfEnabled`? Then you won't need to use pointer on `map[string]interface{}` as you will only modify the existing `map[string]interface{}`
https://play.golang.org/p/Q6vrAmmJWR6

Reply:
BTW Why here we always initialize `in.Labels`, but in Runtime service not?

Reply:
Because in Application creation we're always adding `name` and `integrationSystemID` labels. I put the initilization right above them because it's tied with them, not with adding scenarios (which won't initialize labels if no scenario has to assigned - that happens in runtime creation).